sapi/fpm/fpm/fpm_main.c

```diff
@@ -1836,6 +1836,12 @@ consult the installation file that came with this distribution, or visit \n\
 			/* check if request_method has been sent.
 			 * if not, it's certainly not an HTTP over fcgi request */
 			if (!SG(request_info).request_method) {
+				zend_try {
+					zlog(ZLOG_ERROR, "SCRIPT_FILENAME env not found in cgi env");
```

Comment:
Easy to found problem.
and,I think http 500 is better than 400.because ,this is server's fault.
how do you think?